### ISSUE TITLE
iMain.valueOfTerm containing import returns null after n interactions 

### ISSUE THREAD

**Issue by lresende** [scala/bug#12390]
## reproduction steps

using Scala 2.12.12,

```scala
import scala.tools.nsc.Settings
import scala.tools.nsc.interpreter._

object REPLTest {

  def main(args: Array[String]) = {
    val filename: String = ".../data/generated_01mb.base64"
    val code: String =
        "import scala.io.Source\n " +
        "Source.fromFile(\"" + filename + "\").getLines.mkString(\"\\n\")"

    val s = new Settings()

    s.processArguments(
      List(
        "-Yrepl-class-based",
        "-Yrepl-outdir", "./target"
      ), processAll = true)

    val repl = new IMain(s)
    repl.settings.usejavacp.value = true
    for(i <- 1 to 65) {
      repl.interpret(code) match {
        case IR.Success =>
          val res: String = repl.valueOfTerm(repl.mostRecentVar).get.toString
          println(s"[${i}] - Read ${res.size}")
          assert(repl.valueOfTerm(repl.mostRecentVar).get != null)
      }
    }

  }
}
```

## problem

Constantly start receiving null after the 60th interaction, while the expected behavior would be to continue receiving the value of the most recent processed statement. 

[60] - Read 1333336
import scala.io.Source
res60: String = YWFhYWFhYWFhYWFhYWFhYWFhYWFhYW...
Exception in thread "main" java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:529)
	at scala.None$.get(Option.scala:527)
	at REPLTest$.$anonfun$main$1(driver.scala:25)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
	at REPLTest$.main(driver.scala:22)
	at REPLTest.main(driver.scala)

Note that, separating the execution of the import outside of the loop, you won't see the reported issue and things will complete as expected.

**Comment by SethTisue** [comment]
is the problem reproducible in Scala 2.13.5?

**Comment by som-snytt** [comment]
`JavaMirror.scalaSimpleName` throws on the long class name and `valueOfTerm` ignores the error.

It expects the wrapper class name to have its enclosing class name as a prefix, but some special encoding kicks in at 64 chars:
```
java.lang.AssertionError: assertion failed: Expected class name $line61.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$$$82b5b23cea489b2712a1db46c77e458$$$$w$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw to have prefix $line61.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw
```
The prefix is encroaching on 256 chars.

The underlying issue is that the interpreted imports add unnecessary wrapping. 

That is fixed in 2.13 by a special import directive for REPL that avoids class nesting, though it still issues N such wrappers.

I don't think any of the limitations are likely to progress in 2.12.

**Comment by som-snytt** [comment]
I was going to contribute the assert message but someone already did that 2 years ago

https://github.com/scala/scala/blame/2.13.x/src/reflect/scala/reflect/runtime/JavaMirrors.scala#L1026
